k2/csrc/array_ops.cu

```diff
@@ -128,9 +118,126 @@ Array1<T> Append(int32_t num_arrays, const Array1<T> **src) {
   }
 }
 
+// See documentation in header of what this is supposed to do.
```

Comment:
I assume this was from the .cc file?  Should stay as .cu but dont combine with templated code pls.

Reply:
Yes, I split code in .cc into .cc and .cu. As this make separate compilation dismiss and make cmake smart.